### PR TITLE
Update/7.5 compat request as blog

### LINKED ISSUE #12854: Compat for \Jetpack_Client::wpcom_json_api_request_as_blog($path, $version, $args, $body, $base_api_path)
This is used externally and we should make sure it still works.

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #12854

#### Changes proposed in this Pull Request:

* add compat method

#### Testing instructions:

* Check #12857

#### Proposed changelog entry for your changes:

* Not needed. Part of DNA.

### CODE REVIEW

File: packages/compat/legacy/class.jetpack-client.php [hunk 1]
```diff
@@ -1,4 +1,12 @@
 <?php
+/**
+ * Jetpack Client
+ *
+ * Methods for Jetpack to act as client with wpcom.
```
Comment (dereksmart) [inline packages/compat/legacy/class.jetpack-client.php]:
```suggestion
 * Deprecated methods for Jetpack to act as client with wpcom, provided for back-compatibility.
```

File: packages/compat/legacy/class.jetpack-client.php [hunk 2]
```diff
@@ -10,10 +18,35 @@
...
+	 *
+	 * @deprecated use Automattic\Jetpack\Connection\Client::wpcom_json_api_request_as_blog
+	 *
+	 * @param string $path Endpoint path.
+	 * @param string $version Endpoint version.
+	 * @param array  $args Arguments.
+	 * @param null   $body Request body.
+	 * @param string $base_api_path Endpoint base prefix.
+	 *
+	 * @return \Automattic\Jetpack\Connection\Array|\Automattic\Jetpack\Connection\WP_Error
```
Comment (lezama) [inline packages/compat/legacy/class.jetpack-client.php]:
```suggestion
	 * @return Array|WP_Error
```

File: packages/compat/legacy/class.jetpack-client.php [hunk 3]
```diff
@@ -10,10 +18,35 @@
...
 
 	/**
+	 * Perform remote request.
+	 *
 	 * @deprecated use Automattic\Jetpack\Connection\Client::remote_request
+	 *
+	 * @param array $args Arguments.
+	 * @param null  $body Request body.
+	 *
+	 * @return array|\Automattic\Jetpack\Connection\WP_Error
```
Comment (lezama) [inline packages/compat/legacy/class.jetpack-client.php]:
```suggestion
	 * @return array|WP_Error
```

**Review by gravityrail — APPROVED** [review]
LGTM 🎉

**Pull request merged**